src/App.tsx

```diff
@@ -15,48 +16,129 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  NONE,
+  ALPHABETICALLY,
+  LENGTH,
+}
+
+type ReorderOptions = {
+  sortType: SortType,
+  isReversed: boolean,
+};
+
+function getReorderedGoods(
+  goods: string[],
+  { sortType, isReversed }: ReorderOptions,
+) {
+  const visibleGoods = [...goods];
+
+  visibleGoods.sort((a, b) => {
+    switch (sortType) {
+      case SortType.ALPHABETICALLY:
+        return a.localeCompare(b);
+
+      case SortType.LENGTH:
+        return a.length - b.length;
+
+      case SortType.NONE:
+        return 0;
+
+      default:
+        throw new Error('Error, fixed your sort method');
+    }
+  });
+
+  if (isReversed) {
+    visibleGoods.reverse();
+  }
+
+  return visibleGoods;
+}
+
 export const App: React.FC = () => {
+  const [value, setValue] = useState({
+    isReversed: false,
+    sortType: SortType.NONE,
+  });
+
+  const sortedByAlphabetically = () => {
+    setValue({ ...value, sortType: SortType.ALPHABETICALLY });
+  };
+
+  const sortedByLength = () => {
+    setValue({ ...value, sortType: SortType.LENGTH });
+  };
+
+  const reverse = () => {
+    setValue({ ...value, isReversed: !value.isReversed });
```

Comment:
Combine into one function `sort`, and pass `sortType` and `isReversed` params (also you can add default values for this params).